tests/sys/psa_crypto/main.c

```diff
@@ -22,37 +22,163 @@
 #include "embUnit.h"
 #include "psa/crypto.h"
 
-#define TEST_ASSERT_PSA(func_) TEST_ASSERT_MESSAGE(func_ == PSA_SUCCESS, #func_ " failed");
+void addFailurePSA(const char *func, psa_status_t errcode, long line, const char *file)
+{
+    static char msg[128];
+    strncpy(msg, func, sizeof(msg));
+    strcat(msg, ": ");
+    strcat(msg, psa_status_to_humanly_readable(errcode));
+    addFailure(msg, line, file);
+}
+
+#define TEST_ASSERT_PSA(func_) { psa_status_t ret = func_; if (ret != PSA_SUCCESS) { addFailurePSA(#func_, ret, __LINE__, __FILE__); goto cleanup; } }
 
 /*
  * A second call to psa_crypto_init() should not reset key data.
  */
 static void test_init_twice(void)
 {
-    TEST_ASSERT_PSA(psa_crypto_init());
+    const psa_key_type_t key_type = PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_TWISTED_EDWARDS);
+    const size_t key_bits = 255;
+    const psa_algorithm_t key_alg = PSA_ALG_PURE_EDDSA;
+
+    uint8_t key_data[PSA_EXPORT_PUBLIC_KEY_OUTPUT_SIZE(key_type, key_bits)];
+    size_t key_data_len;
 
+    TEST_ASSERT_PSA(psa_crypto_init());
 
     psa_key_id_t key_id = PSA_KEY_ID_NULL;
     psa_key_attributes_t key_attr = psa_key_attributes_init();
-    psa_set_key_algorithm(&key_attr, PSA_ALG_PURE_EDDSA);
+    psa_set_key_algorithm(&key_attr, key_alg);
     psa_set_key_usage_flags(&key_attr, PSA_KEY_USAGE_EXPORT);
-    psa_set_key_bits(&key_attr, 255);
-    psa_set_key_type(&key_attr, PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_TWISTED_EDWARDS));
+    psa_set_key_bits(&key_attr, key_bits);
+    psa_set_key_type(&key_attr, key_type);
     TEST_ASSERT_PSA(psa_generate_key(&key_attr, &key_id));
 
-    uint8_t key_data[PSA_EXPORT_PUBLIC_KEY_OUTPUT_SIZE(psa_get_key_type(&key_attr), psa_get_key_bits(&key_attr))];
-    size_t key_data_len;
-
     TEST_ASSERT_PSA(psa_export_public_key(key_id, key_data, sizeof(key_data), &key_data_len));
     TEST_ASSERT_PSA(psa_crypto_init());
     TEST_ASSERT_PSA(psa_export_public_key(key_id, key_data, sizeof(key_data), &key_data_len));
 
+cleanup:
+    TEST_ASSERT_PSA(psa_destroy_key(key_id));
+}
+
+/**
+ * Exporting and re-importing a private Ed25519 key should result in the same public key and signature.
+ */
+static void test_export_import_ed25519(void)
```

Comment:
```suggestion
static void test_exported_key_is_identical_when_imported_again_ed25519(void)
```

Not the ideal suggestion but I want to emphasize that this test does NOT test the export and import functions. Instead, it tests integration / inter-operation of those.

If you want to test the functions in general, insert dedicated tests for that. In these test you could then go ahead and answer questions like: What happens when the key parameter is null? Do I get the correct status when the key is bigger than the buffer? etc..

(I am being pedantic here, sorry :angel:)

Reply:
I know the tests are lacking a lot in terms of coverage for PSA Crypto right now. I tried to convey the intent of this specific test with the comment above, but I agree it might be better to use a descriptive function name instead.